### DISCUSSION TITLE
DC Brushed Motors on CF Bolt 1.1

### DISCUSSION THREAD

**Discussion by Marcox2111** [bitcraze/discussions#901] [Q&A]
The question is easy, is there a way to attach DC motors (slightly bigger than cf2.1 ones) to the Bolt without using a dc ESC?

**Comment by tobbeanton** [comment]
Yes there is but it has never really been tested. First the resistors R19, R22, R33, R34 should be mounted with 0R. On the PCB it is the free resistor spot closely together with 2 other resistors on the bottom of the PCB close to the connectors.

Then the motor override signals needs to be disabled and a new motor mapping needs to be added. Let me know if you want help changing that.

**Reply by Marcox2111** [reply]
Yep, I already figured it out. Indeed I was about to write the answer and explain how I did. I general I didn’t had to work at hardware level but what I did was just to swap in the firmware the pin of the motor with the pin of the overide and adjust the pwm accordingly to have a working situation. I’m still testing if this is enough to not have other problems and in case could be good to create a configuration in the Main branch

**Reply by tobbeanton** [reply]
> I general I didn’t had to work at hardware level but what I did was just to swap in the firmware the pin of the motor with the pin of the overide and adjust the pwm accordingly to have a working situation.

Did this work? I'm not sure the override pins are connected to timers? Or maybe I'm not understanding what you did.

**Reply by Marcox2111** [reply]
In the motors_def.c I changed (for the Bolt M1) .gpioPin from GPIO_Pin_1 to GPIO_Pin_0, .gpioPinSource from GPIO_PinSource1 to GPIO_PinSource0 and gpioPowerswitchPin from GPIO_Pin_0 to GPIO_Pin_1,

**Reply by Marcox2111** [reply]
Indeed it was a fortunate coincidence, for M1 it works but not for M2. So I guess that I have to work a little bit deeply to make it work. Should I put the override pin at high impedance?

**Comment by tobbeanton** [comment]
I might find some time to push support for brushed motors on the bolt 1.1 tomorrow but that will require soldering the 0R resistors or a wire to bring the motor signals to the FET. 
![image](https://github.com/bitcraze/discussions/assets/6593186/8830b602-399c-4ece-9376-4d354560dabd)

**Reply by Marcox2111** [reply]
Yes at the end I arrived to this solution too, but is it important to put the pin M1_Override as an input to have an high impedance path otherwise it will always close or open the mosfet. By the way should I create a config file for including supports of brushed motor, (including the pwm modification to work with this mosftet). At the end it works for my project, when I will do the flight test with this platform I will close this discussion.

**Comment by tobbeanton** [comment]
I created a [PR](https://github.com/bitcraze/crazyflie-firmware/pull/1316) that enables brushed support for the Bolt 1.1. Perhaps you can try it? (I've done some basic testing myself so fairly confident it will work)
